plonk-napi/src/types.rs

```diff
@@ -0,0 +1,79 @@
+use kimchi::{linearization::expr_linearization, prover_index::ProverIndex};
+use mina_curves::pasta::{Vesta as GAffine, VestaParameters};
+use mina_poseidon::{constants::PlonkSpongeConstantsKimchi, sponge::DefaultFqSponge};
+use napi::bindgen_prelude::{Error, External, Result as NapiResult, Status, Uint8Array};
+use napi_derive::napi;
+use poly_commitment::ipa::{OpeningProof, SRS};
+use serde::{Deserialize, Serialize};
+use std::{io::Cursor, sync::Arc};
+
+pub struct WasmPastaFpPlonkIndex(pub Box<ProverIndex<GAffine, OpeningProof<GAffine>>>);
+
+// TOOD: remove incl all dependencies when no longer needed and we only pass napi objects around
+#[derive(Serialize, Deserialize)]
+struct SerializedProverIndex {
+    prover_index: Vec<u8>,
```

Comment:
Why does it include separate fields for srs and prover index? In Kimchi's `ProverIndex`, srs is just one of its contents.